lib/livebook/runtime/erl_dist/logger_gl_handler.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Livebook.Runtime.ErlDist.LoggerGLHandler do
+  @moduledoc false
+
+  @doc false
+  def log(%{meta: meta} = event, %{formatter: {formatter_module, formatter_config}}) do
+    message = apply(formatter_module, :format, [event, formatter_config])
+
+    if io_proxy?(meta.gl) do
+      async_io(meta.gl, message)
+    else
+      send(Livebook.Runtime.ErlDist.NodeManager, {:orphan_log, message})
+    end
+  end
+
+  defp io_proxy?(pid) do
+    try do
+      info = Process.info(pid, [:dictionary])
+      info[:dictionary][:"$initial_call"] == {Livebook.Runtime.Evaluator.IOProxy, :init, 1}
```

Comment:
I wonder if we can use the Registry instead of this check... 🤔

Reply:
Will do in a separate PR!